### ISSUE TITLE
Rust 1.19 regression, sliding_windows 2.0.3

### ISSUE THREAD

**Issue by brson** [rust-lang/rust#42464]
https://github.com/flo-l/rust-sliding_windows

```
brian@ip-10-145-43-250:~/dev/slate/rust-sliding_windows⟫ git log -1
commit ff627b0757a43c219d34d888e7705e9570ff904a
Author: flo-l <lacknerflo@gmail.com>
Date:   Thu Mar 17 15:53:32 2016 +0100

    update README

```

```
brian@ip-10-145-43-250:~/dev/slate/rust-sliding_windows⟫ cargo +nightly test
   Compiling sliding_windows v2.0.3 (file:///mnt2/dev/slate/rust-sliding_windows)
    Finished dev [unoptimized + debuginfo] target(s) in 1.22 secs
     Running target/debug/deps/sliding_windows-5ae024d63933df5e

running 6 tests
test tests::sliding_windows_2 ... ok
test tests::sliding_windows_1 ... ok
test tests::sliding_windows_3 ... ok
test tests::sliding_windows_5 ... ok
test tests::sliding_windows_6 ... ok
test tests::sliding_windows_4 ... FAILED

failures:

---- tests::sliding_windows_4 stdout ----
        thread 'tests::sliding_windows_4' panicked at 'assertion failed: `(left == right)` (left: `(18446744073709551610, None)`, right: `(0, None)`)', src/tests.rs:72


failures:
    tests::sliding_windows_4

test result: FAILED. 5 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out

error: test failed, to rerun pass '--lib'
```

cc @flo-l

**Comment by Mark-Simulacrum** [comment]
e0cc22b4bae8007c59fbe58f2c104ecd743d746a -- rollup #42336; https://github.com/rust-lang/rust/pull/42315 which I believe makes this an intentional change.

**Comment by flo-l** [comment]
my test was wrongly assuming this, fixed by flo-l/rust-sliding_windows@1bf7488e4f24de8377acbd687e5382628697adb7

**Comment by flo-l** [comment]
@brson thanks for the heads-up, this issue is resolved and can be closed! :)
